thoth/storages/graph/postgres.py

```diff
@@ -3771,6 +3771,70 @@ def create_python_cve_record(
 
                 return False
 
+    def update_missing_flag_package_version(
+        self,
+        value: bool,
+        index_url: str,
+        package_name: str,
+        package_version: str
+    ) -> None:
+        """Update value of is_missing flag for PythonPackageVersion."""
+        with self._session_scope() as session:
+            subq = (session.query(PythonPackageVersion)
+                    .join(PythonPackageIndex)
+                    .filter(PythonPackageVersion.package_name == package_name)
+                    .filter(PythonPackageVersion.package_version == package_version)
+                    .filter(PythonPackageIndex.url == index_url)
+                    .with_entities(PythonPackageVersion.id)
+                    )
+            (
+                session.query(PythonPackageVersion)
+                .filter(PythonPackageVersion.id.in_(subq))
+                .update({"is_missing": value}, synchronize_session='fetch')
+            )
+
+    def get_all_repositories_using_package(self, index_url: str, package_name: str) -> List[str]:
+        """Given package info, get unique urls of repos which have been advised with this package."""
+        with self._session_scope() as session:
+            result = (
+                session.query(PythonPackageVersion)
+                .join(PythonPackageIndex)
+                .join(PythonRequirementsLock)
+                .join(AdviserRun, AdviserRun.user_software_stack_id == PythonRequirementsLock.python_software_stack_id)
+                .order_by(AdviserRun.origin)
```

Comment:
Is this ordering required? Here and bellow.

Reply:
This one might not be, but the other one definitely is.  `.distinct` gets the first result so we want to make sure that each origin has only one result.  I am trying to ensure that the entry which we get corresponds to the most up to date information which we have on said `origin`.